03_analysis/tailed_parameters.py

```diff
@@ -0,0 +1,537 @@
+    params_id_out : dict
+        same format as params_id_all but with only the subset of outlier mols
+
+    Returns
+    -------
+    nmols_cnt_all : 1D numpy array
+        nmols_cnt_all[i] is the count of molecules that have the parameter
+        found in full_params[i]
+    nmols_cnt_out : 1D numpy array
+        same format as nmols_cnt_all but with only the subset of outlier mols
+    """
+
+    nmols_cnt_all = []
+    nmols_cnt_out = []
+
+    for i, p in enumerate(full_params):
+
+        # count number of mols in the COMPLETE set which use this parameter
+        cnt_all = len(params_id_all[p])
+
+        # count number of mols in the OUTLIER set which use this parameter
+        try:
+            cnt_out = len(params_id_out[p])
+        except KeyError:
+            cnt_out = 0
+
+        nmols_cnt_all.append(cnt_all)
+        nmols_cnt_out.append(cnt_out)
+
+    return np.array(nmols_cnt_all), np.array(nmols_cnt_out)
+
+
+def plot_param_bars(plot_data, labels, max_ratio, suffix, num_sort=False, what_for='talk'):
+    """
+    Generate bar plots of the ratio of (fraction of outlier mols with
+    a given parameter) to (fraction of all mols with a given parameter).
+    The ratios should be pre-computed before calling this function.
+
+    Parameters
+    ----------
+    plot_data : 1D numpy array
+        array of ratios to plot
+    labels : 1D numpy array
+        array of strings that correspond to plot_data
+    max_ratio : float
+        max ratio of all the data that will be plotted to generate consistent
+        plot limits for all outputs
+    suffix: string
+        label to append to end of plot filename
+    num_sort : Boolean
+        True to sort bar plot numerically, False for alphabetical order
+    what_for : string
+        dictates figure size, text size of axis labels, legend, etc.
+        "paper" or "talk"
+    """
+    def single_subplot(ax, y, plot_data, labels):
+        ax.barh(y, plot_data, bar_width, color='darkcyan')
+
+        # add plot labels, ticks, and tick labels
+        ax.set_xlabel('fraction', fontsize=fs1)
+        ax.tick_params(axis='x', labelsize=fs2)
+
+        ax.set_yticks(y)
+        ax.set_yticklabels(labels, fontsize=fs2)
+
+        # invert for horizontal bars
+        ax.invert_yaxis()
+
+        # set plot limits by rounding max_ratio to the nearest 0.5
+        x_max = round(max_ratio * 2) / 2
+        ax.set_xlim(0, x_max)
+        ax.set_xticks(np.linspace(0, x_max, 6))
+
+        # set grid
+        ax.grid(True)
+
+        # add a reference line at ratio = 1.0
+        ax.axvline(1.0, ls='--', c='purple', alpha=0.6)
+
+        # set alternating colors for background for ease of visualizing
+        locs, values = plt.xticks()
+        for i in range(1, len(locs)-1, 2):
+            ax.axvspan(locs[i], locs[i+1], facecolor='lightgrey', alpha=0.25)
+
+    n_bars = len(plot_data)
+    if what_for == 'talk':
```

Comment:
it might be good to allow users to pass a dict in the future which contains these values and allowing passing as a string to hit defaults.